### ISSUE TITLE
"with_children" does not support adding single component

### ISSUE THREAD

**Issue by Raytwo** [bevyengine/bevy#277]
Currently, the implementation requires that the component being spawned implemented DynamicBundle, so you can't do something like this:

```rust
fn setup(
    mut commands: Commands,
) {
    commands
        // Player
        .spawn(SpriteComponents {
            material: materials.add(Color::rgb(0.5, 0.0, 0.0).into()),
            translation: Translation(Vec3::new(window_desc.width as f32 / 2.0, 50.0, 0.0)),
            sprite: Sprite {
                size: Vec2::new(30.0, 30.0),
            },
            ..Default::default()
        })
        .with(Player::new())
        .with_children(|parent| {
            parent.spawn(BulletEmitter::new());
        });
}
```

Instead, you need to do it that way:
```rust
.with_children(|parent| {
            parent.spawn( (BulletEmitter::new(), ) );
        });
}
```

While this is a decent workaround, could it be possible to make it so you can use a regular Component too?

A friend suggested implementing the following, although I'm not knowledgeable about Rust enough to do so myself:
``DynamicBundle for T: Component``

Thanks!

**Comment by ryo33** [comment]
I thought about `.with_child(BulletEmitter::new())`, which wraps the current `with_children`), but it's too specific.

In that case, I prefer to;
- define `BulletEmitterComponents`.
- implement `to_bundle(self)` for `BulletEmitter` if it seems never used with other components.

**Comment by alice-i-cecile** [comment]
This should be solved with the new Commands API in 0.5. See #1703.

**Comment by alice-i-cecile** [comment]
Fixed in 0.5
